### PR TITLE
87 focus item for program

### PR DESCRIPTION
**Pull request by alkenerly**

### CODE REVIEW

File: backend/controllers/program.js [hunk 1]
```diff
@@ -148,3 +148,20 @@ module.exports.deleteOne = (req, res) => {
 			});
 		});
 };
+
+module.exports.getFocusItemByProgram = (req, res) => {
+	FocusItem.find({program: req.params.id})
```
Comment (jvt) [inline backend/controllers/program.js]: We need to do an explicit query on the Program collection. The catch below will only execute on a mongoose failure. It won't catch the "missing program" error

File: backend/routes.js [hunk 2]
```diff
@@ -26,6 +26,7 @@ module.exports = (app, passport) => {
 	 */
 	app.get('/api/programs', [passport.authenticate('jwt', PASSPORT_OPTIONS)], controllers.program.getAll);
 	app.get('/api/program/:id', [passport.authenticate('jwt', PASSPORT_OPTIONS)], controllers.program.getOne);
+	app.get('/api/program/:id/focusitem', [passport.authenticate('jwt', PASSPORT_OPTIONS)], controllers.program.getFocusItemByProgram);
```
Comment (jvt) [inline backend/routes.js]: 👍

File: test/backend/controllers/program.spec.js [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+                .expect(200)
+                
+            
+            
+
+        }
+    });
+
+    after(async function() {
+        this.timeout(15000);
```
Comment (jvt) [inline test/backend/controllers/program.spec.js]: this timeout doesn't do anything?

File: test/backend/controllers/program.spec.js [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+		await Analytic.deleteMany({}).exec();
+        await Teacher.deleteMany({}).exec();
+        await FocusItem.deleteMany({}).exec();
+        await Program.deleteMany({}).exec();
+        await Student.deleteMany({}).exec();
+        return;
```
Comment (jvt) [inline test/backend/controllers/program.spec.js]: don't need to return here

File: test/backend/controllers/program.spec.js [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+        type: 'mobile game'
+    }
+];
+
+describe('Tests the /api/program routes', function() {
+    var teacher;
+    var token;
+
+	before(async function() {
+        this.timeout(15000);
```
Comment (jvt) [inline test/backend/controllers/program.spec.js]: this doesn't do anything?

File: test/backend/controllers/program.spec.js (hunk 3 above)
Comment (alkenerly) [reply test/backend/controllers/program.spec.js]: yeah it does. it'll fail if its not there because it times out

**Pull request merged**